catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPluginImpl.java

```diff
@@ -0,0 +1,138 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.querymonitor;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.Source;
+
+public class QueryMonitorPluginImpl implements QueryMonitorPlugin {
+
+    boolean removeSearchAfterComplete = true;
+
+    private ConcurrentHashMap<UUID, ActiveSearch> activeSearches =
+            new ConcurrentHashMap<UUID, ActiveSearch>();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryMonitorPluginImpl.class);
+
+    public static final String SEARCH_ID = "SEARCH_ID";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ConcurrentHashMap<UUID, ActiveSearch> getActiveSearches() {
+        return activeSearches;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void setRemoveSearchAfterComplete(boolean b) {
+        removeSearchAfterComplete = b;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean addActiveSearch(ActiveSearch as) {
+        if (as == null) {
+            LOGGER.warn("Cannot add null ActiveSearch to map!");
+            return false;
+        }
+        activeSearches.put(as.getUniqueID(), as);
+        return true;
+    }
+
+    public QueryMonitorPluginImpl() {
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean removeActiveSearch(UUID uniqueID) {
+        if (uniqueID == null) {
+            LOGGER.warn("Can't remove active search with null ID.");
+            return false;
+        }
+        return (activeSearches.remove(uniqueID) != null);
+    }
+
+    /**
+     * Method that is implemented for {@link PreFederatedQueryPlugin}. Uses the given {@link Source}
+     * and {@link QueryRequest} information to create a new {@link ActiveSearch} to add to the {@link ActiveSearch} {@link Map}.
+     *
+     * @param source {@link Source} that corresponds to source the search is querying
+     * @param input  {@link QueryRequest} that corresponds to request generated when a user queried the source
+     * @return {@link QueryRequest} that was given as a parameter with updated property information corresponding
+     * to the {@link ActiveSearch}'s {@link UUID}
+     */
+
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        if (source == null) {
+            LOGGER.warn("Source given was null!");
+        }
+        ActiveSearch tempAS = new ActiveSearch(source, input);
+        UUID uniqueID = tempAS.getUniqueID();
+        if (input == null) {
+            LOGGER.error("QueryRequest in process was null! Cannot add active search to map!");
```

Comment:
Most of the DDF logging doesn't use exclamation points, it may be better to use periods to maintain consistency